arrows/klv/apply_child_klv.cxx

```diff
@@ -0,0 +1,252 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// Implementation of apply_child_klv filter.
+
+#include <arrows/klv/apply_child_klv.h>
+
+#include <arrows/klv/klv_0601.h>
+#include <arrows/klv/klv_1607.h>
+#include <arrows/klv/klv_metadata.h>
+
+#include <vital/logger/logger.h>
+#include <vital/range/iterator_range.h>
+
+#include <list>
+
+namespace kwiver {
+
+namespace arrows {
+
+namespace klv {
+
+// ----------------------------------------------------------------------------
+klv_1607_child_policy
+klv_0601_child_policy( klv_lds_key tag )
+{
+  if( klv_0601_traits_lookup().by_tag( tag ).tag_count_range().upper() > 1 )
+  {
+    return KLV_1607_CHILD_POLICY_KEEP_BOTH;
+  }
+  return KLV_1607_CHILD_POLICY_KEEP_CHILD;
+}
+
+// ----------------------------------------------------------------------------
+void
+apply_amend( klv_value& value ) {
+  // Extract the local set
+  auto const set_ptr = value.get_ptr< klv_local_set >();
+  if( !set_ptr )
+  {
+    return;
```

Comment:
Not necessarily; this would only be hit if the value was empty or is just raw bytes due to a parsing error. If it's empty, that's fine and there's no work to do. If it's raw bytes, then there already would have been a message printed when the parsing error happened, and we want to be able to deal with that case without erroring out.

Reply:
Ok, sounds good